database/mongoDb.js

```diff
@@ -0,0 +1,214 @@
+const mongoose = require('mongoose');
+const dbName = 'zillow';
+const config = require('../config');
+
+
+var featuresTable;
+
+const mockFeatures = require('./mockFeatures');
+const mockInterior = require('./mockInterior');
+const features = require('./createMockData');
+const fs = require('fs');
+var documentCount = 0;
+
+
+  var createConnection = function (cb) {
+
+    db = mongoose.connect(`mongodb://${config.host}:27017/${dbName}`, {useNewUrlParser: true});
+    let schema = new mongoose.Schema({
+      house_id: {
+        type: Number,
+        unique: true
+      },
+      type: String,
+      year_built: Number,
+      heating: String,
+      cooling: String,
+      parking: String,
+      lot: Number,
+      days_on_zillow: Number,
+      bedrooms: Number,
+      bathrooms: Number,
+      interiorheating: String,
+      interiorcooling: String,
+      appliances: String,
+      kitchen: String,
+      flooring: String,
+      sqft: Number
+    }, { _id : false });
+    featuresTable = mongoose.model('features', schema);
+    console.log('Connected');
+    featuresTable.countDocuments({}, function(err, count) {
+      if (err) { return handleError(err) } //handle possible errors
+
+      documentCount = count;
+      console.log('The amount of records is: ' + documentCount);
+      cb();

```

Comment:
cb? what does the cb do? Meaningful name please

Reply:
The cb is an empty function.

Why does this exist?

Reply:
Was used when I needed something to happen after the DB was fully opened.  I do have a function sent to it in the 10M DB script where I need to wait to get the amount of records in the DB for Mongo before I start inserting in order to do my own version of an Auto Increment for the House_id field.  Auto incrementing in Mongo was something I had issues with.